premailer/cache.py

```diff
@@ -1,15 +1,43 @@
 import functools
 import os
+import threading
 
-from cachetools import cached, LRUCache
+import cachetools
 
 
-cache = LRUCache(maxsize=int(os.environ.get("PREMAILER_CACHE_MAXSIZE", 128)))
+# Available cache options.
+CACHE_IMPLEMENTATIONS = {
+    "LFU": cachetools.LFUCache,
+    "LRU": cachetools.LRUCache,
+    "RR": cachetools.RRCache,
+    "TTL": cachetools.TTLCache,
```

Comment:
These two; why would you ever want them? Seems like terrible choices. 

Actually, more importantly: Why even give the user the choice to switch to LFU cache? How does it differ from LRU for users? Or does it even matter? If it matters, why isn't one option best for everyone. (thinking about simplicity)

Reply:
I mentioned the usecase where LFU is more suitable than LRU here: https://github.com/peterbe/premailer/issues/225.
I think TTL can still be used in scenarios where after initial parsing, only subset of cache entries are used repetedly, leaving behind entries that are just sitting to consume the memory. Here, TTL can help to clean up those entries after a certain time period.
RR, I agree, doesn't look like it is really useful in this case. Removed it from list of available choices.

Reply:
I see that you switch from `LRUCache({max_size: 128})` to `LFUCache({max_size: 1000})` but that's changing to things at once. Being able to switch from 128 to 1000 was already possible with setting `PREMAILER_CACHE_MAXSIZE` without the need for your hack. 
But why change from LRU to LFU?
And if you've done the research and found that LFU is much better when you're processing thousands of HTML documents, why wouldn't everyone prefer LFU?